OSDOCS#14719: Adding new Azure params (#93547)
openshift/openshift-docs: modules/installation-configuration-parameters.adoc

|====
|Parameter|Description|Values

|compute:
  platform:
    azure:
      bootDiagnostics:
        type:
|This parameter is only for compute machines. `bootDiagnostics` contains the value for the storage account of the Uniform Resource Identifier (URI) that receives the machine log information for the compute machines. `type` specifies the {azure-short} boot diagnostics type for the created compute machines. When you set `type` to `UserManaged`, you must provide the values for `resourceGroup` and `storageAccountName`. When you set `type` to `Managed`, the installation program provides the values for these parameters. When you set `type` to `Disabled`, you turn off the parameter.
|String. For compute machines, the default value is `Disabled`.

|compute:
  platform:
    azure:
      bootDiagnostics:
        resourceGroup:

> |Specifies the name of the {azure-short} group that contains the storage account for diagnostics for compute machines. `resourceGroup` is used only when you set `type` to `UserManaged`.

Reviewer: ```suggestion
|Specifies the name of the {azure-short} resource group that contains the storage account for diagnostics for compute machines. `resourceGroup` is used only when you set `type` to `UserManaged`.
```

|String.

|compute:
  platform:
    azure:
      bootDiagnostics:
        storageAccountName:

> |Specifies the {azure-short} account to store the logs of diagnostics for compute machines. `storageAccountName` is used only where `type` is set to `UserManaged`.

Reviewer: ```suggestion
|Specifies the {azure-short} storage account to store the logs of diagnostics for compute machines. `storageAccountName` is used only where `type` is set to `UserManaged`.
```

|====
|Parameter|Description|Values

|compute:
  platform:
    azure:
      bootDiagnostics:
        type:
|This parameter enables boot diagnostics collection for compute machines. `type` specifies the {azure-short} boot diagnostics type for the created compute machines. When you set `type` to `UserManaged`, you must provide the values for `resourceGroup` and `storageAccountName`. For `storageAccountName` and  {product-title} cluster nodes, ensure that you use the same region and subscription. When you set `type` to `Managed`, {azure-short} stores the data blobs for the boot diagnostics in a managed storage account. When you set `type` to `Disabled`, you turn off the parameter.
|String. For compute machines, the default value is `Disabled`.

|compute:
  platform:
    azure:
      bootDiagnostics:
        resourceGroup:

> |Specifies the name of the {azure-short} resource group that contains the storage account for diagnostics for compute machines. `resourceGroup` is used only when you set `type` to `UserManaged`.

Reviewer 2: ```suggestion
|Specifies the name of the {azure-short} resource group that contains the storage account for diagnostics for compute machines. Use `resourceGroup` only when you set `type` to `UserManaged`.
```
https://www.ibm.com/docs/en/ibm-style?topic=grammar-verbs#voice - use active voice

|String.

|compute:
  platform:
    azure:
      bootDiagnostics:
        storageAccountName:

> |Specifies the {azure-short} storage account to store the logs of diagnostics for compute machines. `storageAccountName` is used only where `type` is set to `UserManaged`.

Reviewer 2: ```suggestion
|Specifies the {azure-short} storage account to store the logs of diagnostics for compute machines. use `storageAccountName` only when you set `type` to `UserManaged`.
```
https://www.ibm.com/docs/en/ibm-style?topic=grammar-verbs#voice - use active voice

Supplying more than one user-assigned identity is an experimental feature, which may be enabled with the `MachineAPIMigration` feature gate.
|Array of strings.

|controlPlane:
  platform:
    azure:
      bootDiagnostics:
        resourceGroup:

> |Specifies the name of the {azure-short} resource group that contains the storage account for diagnostics for control plane machines. `resourceGroup` is used only when you set `type` to `UserManaged`.

Reviewer 2: ```suggestion
|Specifies the name of the {azure-short} resource group that contains the storage account for diagnostics for control plane machines. Use `resourceGroup` only when you set `type` to `UserManaged`.
```
https://www.ibm.com/docs/en/ibm-style?topic=grammar-verbs#voice - use active voice

|The availability zones where the installation program creates control plane machines.
|String list

|platform:
  azure:
    defaultMachinePlatform:
       bootDiagnostics:
        type:
|This parameter enables boot diagnostics collection for all machines. `type` specifies the {azure-short} boot diagnostics type for all the created machines. When you set `type` to `UserManaged`, you must provide the values for `resourceGroup` and `storageAccountName`. When you set `type` to `Managed`, the installation program provides the values for these parameters. When you set `type` to `Disabled`, you turn off the parameter.

> |String. For all machines, the default value is `Disabled`.

Author: @jinyunma, Thanks for your review. PTAL at the updated PR.

|The availability zones where the installation program creates control plane machines.
|String list

|platform:
  azure:
    defaultMachinePlatform:
       bootDiagnostics:
        type:

> |This parameter enables boot diagnostics collection for all machines. `type` specifies the {azure-short} boot diagnostics type for all the created machines. When you set `type` to `UserManaged`, you must provide the values for `resourceGroup` and `storageAccountName`. For `storageAccountName` and {product-title} cluster nodes, ensure that you use the same region and subscription. When you set `type` to `Managed`, {azure-short} stores the data blobs for the boot diagnostics in a managed storage account. When you set `type` to `Disabled`, you turn off the parameter.

Reviewer 2: ```suggestion
|Enables boot diagnostics collection for all machines. The `type` field specifies the {azure-short} boot diagnostics type for all the created machines. When you set `type` to `UserManaged`, you must provide values for `resourceGroup` and `storageAccountName`. For `storageAccountName` and {product-title} cluster nodes, use the same region and subscription. When you set `type` to `Managed`, {azure-short} stores the boot diagnostics data blobs in a managed storage account. When you set `type` to `Disabled`, you turn off the parameter.
```
Suggestion to remove "This parameter" to maintain consistency with other descriptions.
Rewording for clarity.

|String.

|platform:
  azure:
    defaultMachinePlatform:
       bootDiagnostics:
        resourceGroup:
|Specifies the name of the {azure-short} resource group that contains the storage account for diagnostics for all machines. `resourceGroup` is used only when you set `type` to `UserManaged`.
|String.

|platform:
  azure:
    defaultMachinePlatform:
       bootDiagnostics:
        storageAccountName:

> |Specifies the {azure-short} storage account to store the logs of diagnostics for all machines. `storageAccountName` is used only where `type` is set to `UserManaged`.

Reviewer 2: ```suggestion
|Specifies the {azure-short} storage account to store the logs of diagnostics for all machines. Use `storageAccountName` only when you set `type` to `UserManaged`.
```
